sdk/core/core-rest-pipeline/src/util/tokenCycler.ts

```diff
@@ -201,9 +201,11 @@ export function createTokenCycler(
     //   step 1.
     //
 
-    // IF the tenantId passed in token options is different to the one we have, we need to
-    // refresh the token with the new tenantId.
-    const mustRefresh = tenantId !== tokenOptions.tenantId || cycler.mustRefresh;
+    // If the tenantId passed in token options is different to the one we have
+    // Or if we are in claim challenge and the token was rejected and a new access token need to be issued, we need to
+    // refresh the token with the new tenantId or token.
+    const mustRefresh =
+      tenantId !== tokenOptions.tenantId || Boolean(tokenOptions.claims) || cycler.mustRefresh;
```

Comment:
so whenever we see claims, we know we have to refresh? Can we add a test with claims and one without to make sure those booleans are covered?

Reply:
Sure will add a test to cover this